src/DurableTask.AzureStorage/AzureStorageOrchestrationService.cs

```diff
@@ -837,6 +845,12 @@ async Task AbandonAndReleaseSessionAsync(OrchestrationSession session)
             try
             {
                 session.ETag = await this.trackingStore.UpdateStateAsync(runtimeState, instanceId, executionId, session.ETag);
+
+                if(this.trackingStore is InstanceStoreBackedTrackingStore && workItem.OrchestrationRuntimeState != runtimeState){
```

Comment:
nit: minor formatting updates
```suggestion
                if (this.trackingStore is InstanceStoreBackedTrackingStore && workItem.OrchestrationRuntimeState != runtimeState)
                {
```